src/DesktopPixmap.cpp

```diff
@@ -0,0 +1,99 @@
+#include "DesktopPixmap.hpp"
+#include <iostream>
+#include <ostream>
+#include <xcb/xcb.h>
+#include <xcb/xproto.h>
+
+DesktopPixmap::DesktopPixmap (
+    const short x,          ///< x coordinate of the top left corner
+    const short y,          ///< y coordinate of the top left corner
+    const uint16_t height,  ///< height of display
+    const uint16_t width,   ///< width of display
+    const std::string &name ///< _NET_WM_NAME of display
+)
+{
+  this->x = x;
+  this->y = y;
+  this->height = height;
+  this->width = width;
+  this->name = name; // XXX Is this OK?
+
+  // Initializing non built-in types to zeros
+  DesktopPixmap::gc_ = 0;
+  this->image_ptr = nullptr;
+  this->pixmap_id = 0;
+  this->length = 0;
+}
+
+DesktopPixmap::~DesktopPixmap ()
+{
+  delete[] this->image_ptr;
+  // TODO Might need to delete pixmap
```

Comment:
You can set it free via xcb_free_pixmap anytime, I think. If it's not what you mean by delete, then what?

Reply:
Yes, you are right. Will have to free pixmap in the destructor

Reply:
If I do `xcb_disconnect`, will every x object allocated on server be freed?
Sounds reasonable